diff --git a/gfx/context/mali_fbdev_ctx.c b/gfx/context/mali_fbdev_ctx.c
index 841948f..5f8dc3f 100644
--- a/gfx/context/mali_fbdev_ctx.c
+++ b/gfx/context/mali_fbdev_ctx.c
@@ -199,6 +199,12 @@ static bool gfx_ctx_set_video_mode(void *data,
       EGL_NONE
    };
 
+   typedef struct fbdev_window
+	{
+	unsigned short width;
+	unsigned short height;
+	} fbdev_window;
+
    struct fbdev_window window = { width, height };
    if ((g_egl_surf = eglCreateWindowSurface(g_egl_dpy, g_config, &window, 0)) == EGL_NO_SURFACE)
    {
diff --git a/qb/config.libs.sh b/qb/config.libs.sh
index 8ee553c..16ab286 100644
--- a/qb/config.libs.sh
+++ b/qb/config.libs.sh
@@ -191,7 +191,7 @@ fi
 check_lib DYNAMIC "$DYLIB" dlopen
 
 if [ "$HAVE_KMS" != "no" ]; then
-   check_pkgconf GBM gbm 9.0
+   check_pkgconf GBM gbm
    check_pkgconf DRM libdrm
    if [ "$HAVE_GBM" = "yes" ] && [ "$HAVE_DRM" = "yes" ] && [ "$HAVE_EGL" = "yes" ]; then
       HAVE_KMS=yes
